rust/rbac-registration/examples.md

```diff
@@ -148,45 +164,90 @@ The existing chain can be updated instead.
+    style D0 fill: green
+    style AF fill: gray
+    style CF fill: green
+    style DF fill: green
 ```
 
 In the example above there is the `Chain1` registration chain that was created with some `Stake1` stake address and
 `PubKey1` key then both address and key were updated to `Stake2` and `PubKey2`.
+
 The `Chain2` registration is invalid because it uses the same stake address and key.
 There are no reason to start a new registration chain with the same data already registered and not to update the
 existing one, so this isn't allowed.
 
 The `Chain3` registration is valid because the `PubKey2` key wasn't used before to start a chain and the `Stake1` stake
 address isn't currently used by `Chain1`.
-Warning: this scenario is currently problematic because we never remove entries from the `catalyst_id_for_stake_address`
-table.
-Therefore, when registration information is requested using the `Stake1` address we need to build both chains to
-determine that only `Chain3` is relevant.
 
-The `Chain4` registration is valid, but it "overrides" `Chain1` essentially discarding it and starting a new chain
-because it has the same stake address (`Stake2`).
+The `Chain4` registration is valid, but it takes ownership over the `Stake2` stake address of the first chain.
+It is allowed because the new chain uses a new public key.
 This can be useful if a user lost his private key while maintaining access to his Cardano wallet (and the stake address
 used in that registration chain).
-After that `Chain1` is considered no longer valid and must not be used.
+This can also be referred as "restarting" or "overriding" a chain.
+After that `Chain1` is still valid and can be potentially continued, but as for now it has zero voting power because it
+has no active stake address.
+
+### Restarting multiple chains
+
+```mermaid
+block-beta
+    columns 6
+    space s1["slot 1"] s2["slot 2"] s3["slot 3"] space res["result"]
+    c1["chain 1"] A0["..."] A1["Stake1\nPubKey1"] space space AF["\nPubKey1"]
+    c2["chain 2"] B0["..."] B1["Stake2\nPubKey2"] space space BF["\nPubKey2"]
+    c3["chain 3"] space space C0["Stake1, Stake2\nPubKey3"] space CF["Stake1, Stake2\nPubKey2"]
+    style A0 fill: green
+    style A1 fill: green
+    style AF fill: gray
+    style B0 fill: green
+    style B1 fill: green
+    style BF fill: gray
+    style C0 fill: green
+    style CF fill: green
+```
+
+It is possible to restart/override multiple chains with one new registration.
+In the example above there are two chains with stake addresses `Stake1` and `Stake2` correspondingly.
+The new root registration contains both of these addresses, so assuming it is valid, it "restarts" both two previous
+registrations.
+
+As explained in the [restarting a chain](#restarting-a-chain) example, a new registration must use a different public
+key, so using both `PubKey1` and `PubKey2` isn't allowed.
+
+Both first and second chains are still valid and can be continued, but have zero voting power.
```

Comment:
Should we mention that It can still perform basic actions required by the application?

Reply:
Good idea!